app/src/main/java/org/p2p/wallet/newsend/ui/linkgeneration/SendLinkGenerationPresenter.kt

```diff
@@ -0,0 +1,36 @@
+package org.p2p.wallet.newsend.ui.linkgeneration
+
+import timber.log.Timber
+import java.math.BigInteger
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.p2p.core.token.Token
+import org.p2p.core.utils.fromLamports
+import org.p2p.wallet.BuildConfig
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.newsend.interactor.SendInteractor
+import org.p2p.wallet.newsend.model.LinkGenerationState
+import org.p2p.wallet.newsend.model.TemporaryAccount
+
+class SendLinkGenerationPresenter(
+    private val sendInteractor: SendInteractor
+) : BasePresenter<SendLinkGenerationContract.View>(),
+    SendLinkGenerationContract.Presenter {
+
+    override fun generateLink(recipient: TemporaryAccount, token: Token.Active, lamports: BigInteger) {
```

Comment:
lamports  -> tokenAmount ?

tokenAmount.fromLamports looks nice

Reply:
you must be kidding. 
The lamports are passed to fee relayer and we need the exact amount in lamports there

Reply:
no no, i mean renaming the variable
it's just from the name i couldn't figure out what lamports are you passing here